src/site/content/en/blog/fetch-upload-streaming/index.md

```diff
@@ -0,0 +1,359 @@
+const responsePromise = fetch(url, {
+  method: 'POST',
+  body: readableStream,
+});
+```
+
+The next best thing to duplex communication is to make one fetch with a
+streaming request, then make another fetch to receive the streaming response.
+The server will need some way to associate these two requests, like an ID in the
+URL. That's how the [demo](#demo) works.
+
+## Potential issues
+
+Yeah, so… this is a new feature, and one that's underused on the internet today.
+Here are some issues to look out for:
+
+### Incompatibility on the server side
+
+Some app servers don't support streaming requests, and instead wait for the full
+request to be received before letting you see any of it, which kinda defeats the
+point. Instead, use an app server that supports streaming, like
+[NodeJS](https://nodejs.org/dist/latest-v14.x/docs/api/http.html#http_class_http_incomingmessage).
+
+But, you're not out of the woods yet! The application server, such as NodeJS,
+usually sits behind another server, often called a "front-end server", which may
+in turn sit behind a CDN. If any of those decide to buffer the request before
+giving it to the next server in the chain, you lose the benefit of request
+streaming.
+
+Also, if you're using HTTP/1.1, one of the servers may not be prepared for
+chunked encoding, and may fail with an error. But hey, at least you can test
+that and try to change servers if needed.
+
+_…long sigh…_
+
+### Incompatibility outside of your control
+
+If you're using HTTPS you don't need to worry about proxies between you and the
+user, but the user may be running a proxy on their machine. Some internet
+protection software does this to allow it to monitor everything that goes
+between the browser and network.
+
+There may be cases where this software buffers request bodies, or in the case of
+HTTP/1.1, doesn't expect chunked encoding, and breaks in some exciting way.
+
+Right now, it's not clear how often this will happen, if at all.
+
+If you want to protect against this, you can create a 'feature test' similar to
+the [demo above](#demo), where you try to stream some data without closing the
+stream. If the server receives the data, it can respond via a different fetch.
+Once this happens, you know the client supports streaming requests end-to-end.
+
+## Feedback welcome
+
+Feedback from the community is crucial to the design of new APIs, so please try
+it out and tell us what you think! If you encounter any bugs, please [report
+them](https://bugs.chromium.org/p/chromium/issues/list), but if you have general
+feedback, please send it to the [blink-network-dev Google
+Group](https://groups.google.com/a/chromium.org/forum/#!forum/blink-network-dev).
+
+Photo by [Laura
+Lefurgey-Smith](https://unsplash.com/@lauralefurgeysmith?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText)
+on
+[Unsplash](https://unsplash.com/s/photos/canoe-stream?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText)
```

Comment:
Is this a copy/paste error? The image doesn't appear on the page

Reply:
I forgot to add it to the metadat. I fixed it.